pkg/storage/namedstatements/cache.go

```diff
@@ -0,0 +1,79 @@
+package namedstatements
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/jmoiron/sqlx"
+)
+
+// Cache is a cache of prepared statements.
+type Cache struct {
+	db                 *sqlx.DB
+	preparedStatements map[string]*sqlx.NamedStmt
+	lock               sync.RWMutex
+}
+
+// NewCache initializes a new Cache instance with the provided database.
+func NewCache(db *sqlx.DB) *Cache {
+	return &Cache{
+		db:                 db,
+		preparedStatements: make(map[string]*sqlx.NamedStmt),
+	}
+}
+
+// Get returns a prepared statement for the provided query.
+func (n *Cache) Get(query string) (*sqlx.NamedStmt, error) {
```

Comment:
nitpick, but `c` makes more sense to me for these, since we're dealing with a `*Cache`
```suggestion
func (c *Cache) Get(query string) (*sqlx.NamedStmt, error) {
```